compiler/rustc_ty_utils/src/common_traits.rs

```diff
@@ -33,6 +34,26 @@ fn is_item_raw<'tcx>(
     traits::type_known_to_meet_bound_modulo_regions(&infcx, param_env, ty, trait_def_id)
 }
 
+fn type_is_uninhabited_from_raw<'tcx>(
```

Comment:
Since we're already caching this for non-infer tys, do we *need* to do caching in the (old) trait solver as well?